Rajitha/swapSubstring.js

```diff
@@ -0,0 +1,49 @@
+/*****************************************************************************
+ * Swapping Two by Two
+ * Write a function that swaps the first pair (1st and 2nd characters) with
+ * the second pair (3rd and 4th characters) for every quadruplet substring.
+ * Examples
+ * swapTwo("ABCDEFGH") ➞ "CDABGHEF"
+ * swapTwo("AABBCCDDEEFF") ➞ "BBAADDCCFFEE"
+ * swapTwo("munchkins") ➞ "ncmuinhks"
+ * swapTwo("FFGGHHI") ➞ "GGFFHHI"
+ * Notes
+ * Keep leftover strings in the same order.
+ ****************************************************************************/
+const swapTwo = str => {
+  let strLength = str.length;  
+  let iterate = Math.trunc(strLength / 4);
+  let leftOver = strLength % 4;
+  let tempStr;
+  let outputStr = "";
+
+  for (let i = 0; i < iterate; i++){
+    tempStr = str.substr(0,4);
```

Comment:
why is this inefficient?

Reply:
because 5th char in str doesn't exist in case of last iteration?